### PR TITLE
LPS-142849 Use the field label in the user interface

### PR DESCRIPTION
**Pull request by robertoDiaz**
Hi @liferay-forms team.

What do you think about this fix?

If you wish I could generify it and apply for all the exception types.

Thanks!!

### CODE REVIEW

**Comment by liferay-forms** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
Hi, you don’t need to run the CI commands anymore, I’ll do it for you and let you know if any issue happens.

**Comment by liferay-forms** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/display/view.jsp [hunk 1]
```diff
@@ -128,9 +128,17 @@ long formInstanceId = ddmFormDisplayContext.getFormInstanceId();
 
 							<%
 							DDMFormValuesValidationException.MustSetValidValue msvv = (DDMFormValuesValidationException.MustSetValidValue)errorException;
+
+							String fieldName = msvv.getFieldName();
+
+							LocalizedValue fieldLabel = msvv.getFieldLabel();
+
+							if (fieldLabel != null) {
+								fieldName = fieldLabel.getString(themeDisplay.getLocale());
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/display/view.jsp:137]: @robertoDiaz I was wondering if we could check if `fieldLabel.getString(themeDisplay.getLocale())` is empty and then use `fieldName` if it is. What do you think?

Comment (robertoDiaz) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/display/view.jsp:137]: sounds good. 👍 

Anyway, what do you think about generify this? (apply it to MustSetValidValuesSize and more too)

Comment (carolmariaabb) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/display/view.jsp:137]: I do not see a problem. How many would it be?

Comment (robertoDiaz) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/display/view.jsp:137]: Most of the DDMFormValuesValidationException inner exceptions (the ones that follow the same pattern as MustSetValidValue and receive the fieldName).

I'll do this changes and send it to you in separate commits.

Thanks!!

**Comment by robertoDiaz** [comment]
ci:test:sf

**Comment by robertoDiaz** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @robertoDiaz, can you please check if these failures are related to your changes? Thanks

**Comment by robertoDiaz** [comment]
Hi.

It seems that https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0/505335//testReport/com.liferay.poshi.runner/PoshiRunner/test_LocalFile_FormsTextField_CanBeValidatedUsingTheDoesNotContainOption_ could be related somehow , but I do not know how.

My changes are related with validations and it seems that this functional test is detecting (if I understand it well) a problem with the title field when editing:

From poshi report:

`Element is not present at "//div[contains(@class,'ddm-form-basic-info')]//div//div[contains(@class,'form-group')]//textarea[contains(@id, 'nameEditor')]"`

I've check the resulting html in my local and this is the result:

`<div class="form-group"><textarea class="ddm-form-description form-control" id="descriptionEditor" name="_com_liferay_dynamic_data_mapping_form_web_portlet_DDMFormAdminPortlet_descriptionEditor" placeholder="Add a short description for this form." style="height: 24px;"></textarea></div>`

Could your QA tester share with me what is the expected result for this test?

Thanks!!

**Review by carolmariaabb — APPROVED** [review]
Hi @robertoDiaz,

Regarding the test, it's actually a flaky. Please ignore it.

Thanks!

**Comment by carolmariaabb** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:forward

**Pull request closed without merging**